src/ipc/ipc4/helper.c

```diff
@@ -622,7 +621,7 @@ int ipc4_trigger_chain_dma(struct ipc *ipc, struct ipc4_chain_dma *cdma)
 {
 	struct ipc_comp_dev *ipc_pipe;
 	uint32_t pipeline_id;
-	int ret;
+	int ret = -EINVAL;
```

Comment:
This doesn't seem right. This function returns error codes like `IPC4_INVALID_RESOURCE_ID`, not POSIX values. So we probably never want to return `-EINVAL` from it? OTOH it also **almost** looks like the only value we want to return is `IPC4_SUCCESS` a.k.a. 0 in the end. So it **looks** like we just could do `return IPC4_SUCCESS;` in the end - but only almost. Because there's one case in this function in line 652 / 651 below where `ret` is assigned an error code instead of being returned immediately. But in fact that line can also be replaced with a simple `return` and the conditions could be simplified too... Hmm, how about #5234 instead of this patch?

Reply:
I would recommend taking #5234 as well instead of this change.